packages/tpetra/CMakeLists.txt

```diff
@@ -1159,6 +1160,39 @@ GLOBAL_SET(${PACKAGE_NAME}_ETI_GORDS   ${${PACKAGE_NAME}_ETI_GORDS})
 GLOBAL_SET(${PACKAGE_NAME}_ETI_DEVICES ${${PACKAGE_NAME}_ETI_DEVICES})
 GLOBAL_SET(${PACKAGE_NAME}_ETI_NODES   ${${PACKAGE_NAME}_ETI_NODES})
 
+#
+# Verify that a valid GORDS configuration is set.
+#
+MESSAGE(STATUS "")
+MESSAGE(STATUS "Tpetra: Validate global ordinal setting ...")
+SET(lenTpetraCore_ETI_GORDS 0 INTERNAL)
+LIST(LENGTH ${PACKAGE_NAME}_ETI_GORDS lenTpetraCore_ETI_GORDS)
+
+IF(Tpetra_ENABLE_DEPRECATED_CODE)
+    IF(lenTpetraCore_ETI_GORDS EQUAL 0)
+        MESSAGE(FATAL_ERROR "Tpetra requires a global ordinal type, but none were specified. "
```

Comment:
This check makes lines 714-727 (see above) redundant and unnecessary.  You could actually delete that check above in favor of your new check here.  The redundant check above won't break anything, so it's OK to address that in a future PR.